Upstream: cherry-picked from master

From 154c16dc80edf5e0ee459ad18442c43fc0a2ed1b Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Wed, 6 Aug 2014 16:40:45 +0200
Subject: [PATCH] Add an option to only build baloo's libraries

Adds the BALOO_LIBRARIES_ONLY cmake option to not build any of the
executables.

REVIEW: 118438

Conflicts:
	CMakeLists.txt
	src/CMakeLists.txt
	src/file/CMakeLists.txt
---
 CMakeLists.txt          |   2 +
 src/CMakeLists.txt      |  10 ++--
 src/file/CMakeLists.txt | 118 ++++++++++++++++++++++++------------------------
 3 files changed, 68 insertions(+), 62 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ba999a..99206e6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,8 @@ set(BALOO_VERSION "${BALOO_VERSION_MAJOR}.${BALOO_VERSION_MINOR}.${BALOO_VERSION
 
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules" ${CMAKE_MODULE_PATH})
 
+option(BALOO_LIBRARIES_ONLY "If true, compiles only the Baloo libraries, without binaries, kioslaves." OFF)
+
 find_package(KDE4 REQUIRED)
 include(KDE4Defaults)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3312553..904bf40 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -2,7 +2,9 @@ add_subdirectory(core)
 add_subdirectory(xapian)
 add_subdirectory(file)
 add_subdirectory(pim)
-add_subdirectory(kioslaves)
-add_subdirectory(tools)
-add_subdirectory(runner)
-add_subdirectory(queryparser)
+if (NOT BALOO_LIBRARIES_ONLY)
+    add_subdirectory(kioslaves)
+    add_subdirectory(tools)
+    add_subdirectory(runner)
+    add_subdirectory(queryparser)
+endif ()
diff --git a/src/file/CMakeLists.txt b/src/file/CMakeLists.txt
index 49a34dd..8f93038 100644
--- a/src/file/CMakeLists.txt
+++ b/src/file/CMakeLists.txt
@@ -3,69 +3,71 @@ add_definitions(${KDE4_ENABLE_EXCEPTIONS})
 add_subdirectory(lib)
 add_subdirectory(search)
 
-set(file_SRCS
-    # Filewatch
-    filewatch.cpp
-    activefilequeue.cpp
-    metadatamover.cpp
-    database.cpp
-
-    # File Indexer
-    fileindexer.cpp
-    fileindexerconfig.cpp
-    indexscheduler.cpp
-    indexingqueue.cpp
-    basicindexingqueue.cpp
-    fileindexingqueue.cpp
-    commitqueue.cpp
-    basicindexingjob.cpp
-    fileindexingjob.cpp
-    eventmonitor.cpp
-    util.cpp
+if (NOT BALOO_LIBRARIES_ONLY)
+    set(file_SRCS
+        # Filewatch
+        filewatch.cpp
+        activefilequeue.cpp
+        metadatamover.cpp
+        database.cpp
 
-    # Common
-    priority.cpp
-    regexpcache.cpp
-    fileexcludefilters.cpp
-    # removabledeviceindexnotification.cpp
-    # removablemediacache.cpp
+        # File Indexer
+        fileindexer.cpp
+        fileindexerconfig.cpp
+        indexscheduler.cpp
+        indexingqueue.cpp
+        basicindexingqueue.cpp
+        fileindexingqueue.cpp
+        commitqueue.cpp
+        basicindexingjob.cpp
+        fileindexingjob.cpp
+        eventmonitor.cpp
+        util.cpp
 
-    main.cpp
-)
+        # Common
+        priority.cpp
+        regexpcache.cpp
+        fileexcludefilters.cpp
+        # removabledeviceindexnotification.cpp
+        # removablemediacache.cpp
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-    set(file_SRCS
-        ${file_SRCS}
-        kinotify.cpp
+        main.cpp
     )
-    add_definitions(-DBUILD_KINOTIFY)
-endif()
 
-kde4_add_executable(baloo_file ${file_SRCS})
+    if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+        set(file_SRCS
+            ${file_SRCS}
+            kinotify.cpp
+        )
+        add_definitions(-DBUILD_KINOTIFY)
+    endif()
 
-target_link_libraries(baloo_file
-    ${QT_QTCORE_LIBRARY}
-    ${QT_QTSQL_LIBRARY}
-    ${KDE4_KDECORE_LIBS}
-    ${KDE4_KIO_LIBS}
-    ${KDE4_KDEUI_LIBS}
-    ${KDE4_KIDLETIME_LIBS}
-    ${KDE4_SOLID_LIBS}
-    ${KFILEMETADATA_LIBRARY}
-    ${XAPIAN_LIBRARIES}
-    ${QJSON_LIBRARIES}
-    baloocore
-    baloofiles
-    balooxapian
-)
+    kde4_add_executable(baloo_file ${file_SRCS})
 
-install(TARGETS baloo_file ${INSTALL_TARGETS_DEFAULT_ARGS})
-install(FILES baloo_file.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
-install(FILES org.kde.baloo.file.indexer.xml DESTINATION ${DBUS_INTERFACES_INSTALL_DIR})
+    target_link_libraries(baloo_file
+        ${QT_QTCORE_LIBRARY}
+        ${QT_QTSQL_LIBRARY}
+        ${KDE4_KDECORE_LIBS}
+        ${KDE4_KIO_LIBS}
+        ${KDE4_KDEUI_LIBS}
+        ${KDE4_KIDLETIME_LIBS}
+        ${KDE4_SOLID_LIBS}
+        ${KFILEMETADATA_LIBRARY}
+        ${XAPIAN_LIBRARIES}
+        ${QJSON_LIBRARIES}
+        baloocore
+        baloofiles
+        balooxapian
+    )
+
+    install(TARGETS baloo_file ${INSTALL_TARGETS_DEFAULT_ARGS})
+    install(FILES baloo_file.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
+    install(FILES org.kde.baloo.file.indexer.xml DESTINATION ${DBUS_INTERFACES_INSTALL_DIR})
 
-add_subdirectory(autotest)
-add_subdirectory(tests)
-add_subdirectory(extractor)
-add_subdirectory(kcm)
-add_subdirectory(inotify)
-add_subdirectory(cleaner)
+    add_subdirectory(autotest)
+    add_subdirectory(tests)
+    add_subdirectory(extractor)
+    add_subdirectory(kcm)
+    add_subdirectory(inotify)
+    add_subdirectory(cleaner)
+endif()
-- 
2.1.0

